tools/python_tools/cuvslam_tools/dataset_preparation/tum/prepare.py

```
@@ -0,0 +1,137 @@
+
+    dataset_dir = output_dir / DATASET_NAME
+    sequence_dir = dataset_dir / SEQUENCE_NAME
+    archive = raw_dir / f"{SEQUENCE_NAME}.tgz"
+
+    print()
+    print(f"Extracting {archive.name} …")
+    dataset_dir.mkdir(parents=True, exist_ok=True)
+    extract_archive(archive, dataset_dir)
+
+    if not sequence_dir.is_dir():
+        raise PreparationError(f"expected {sequence_dir} after extraction")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)